packages/lib-bitcoinjs/index.ts

```diff
@@ -81,6 +119,11 @@ export class DfnsWallet implements SignerAsync {
 
     return Psbt.fromHex(res.signedData.replace(/^0x/, ''))
   }
+
+  public tweakTaprootMerkleRoot(taprootMerkleRoot: Buffer) {
+    this.taprootMerkleRoot = taprootMerkleRoot
+    this.publicKey = getTapInternalKey(this.rootPublicKey, this.taprootMerkleRoot, true)
```

Comment:
nit: instead of setting `this.publicKey` in different places, we could implement it as a getter maybe ? which depends on `this.rootPublicKey` / `this.scheme` / `this. taprootMerkleRoot`

Reply:
unfortunately we have `export class DfnsWallet implements SignerAsync`
this object is then passed to the bitcoin library, it needs to call `publicKey`.

An alternative would be to clone when tweaking, so it would be something like:

```
const wallet = = await initDfnsWallet(process.env.BITCOIN_WALLET_ID!)
const wallet2 = wallet.tweak(taprootMerkleRoot)
```

Reply:
hmm, I don't like the mutating state inside the wallet, but also doesn't make sense to model the tweak as a separate wallet, the underlying wallet is the same. you are just tweaking the signing. the way I would implement this is leave the `DfnsWallet` to only sign with the root key, or tweaked root key for schnorr, introduce a new function `tweakSigner` to return an tweaked signer as an anonymous class for signing with a different merkle root.

```
class DfnsWallet {
 ...

  public tweakSigner(taprootMerkleRoot: Buffer): SignerAsync {
    const tweakedKey = tweakKey(this.publicKey, taprootMerkleRoot)

    return new class implements SignerAsync {
      get publicKey() {
        return tweakedKey
      }

      async signSchnorr(hash: Buffer): Promise<Buffer> {
        // sign with merkle root
      }
    }
  }
}


const tweakedSigner = wallet.tweakSigner(taprootMerkleRoot)
await psbt.signInputAsync(1, tweakedSigner)
```